mpmp/prediction/regression.py

```diff
@@ -0,0 +1,128 @@
+"""
+Functions for training regression models on TCGA data.
+
+"""
+import numpy as np
+import pandas as pd
+from sklearn.pipeline import Pipeline
+from sklearn.linear_model import ElasticNet
+from sklearn.metrics import mean_squared_error, r2_score
+from sklearn.model_selection import (
+    cross_val_predict,
+    GridSearchCV,
+    RandomizedSearchCV,
+)
+
+def train_regressor(X_train,
+                    X_test,
+                    y_train,
+                    alphas,
+                    l1_ratios,
+                    seed,
+                    n_folds=4,
+                    max_iter=1000):
+    """
+    Build the logic and sklearn pipelines to predict real-valued y from dataset x
+
+    Arguments
+    ---------
+    X_train: pandas DataFrame of feature matrix for training data
+    X_test: pandas DataFrame of feature matrix for testing data
+    y_train: pandas DataFrame of processed real-valued labels
+    alphas: list of alphas to perform cross validation over
+    l1_ratios: list of l1 mixing parameters to perform cross validation over
+    n_folds: int of how many folds of cross validation to perform
+    max_iter: the maximum number of iterations to test until convergence
+
+    Returns
+    ------
+    The full pipeline sklearn object and y matrix predictions for training,
+    testing, and cross validation
+    """
+    # Setup the regression parameters
+    import mpmp.config as cfg
+    reg_parameters = {
+        "regress__alpha": alphas,
+        "regress__l1_ratio": l1_ratios,
+    }
+
+    estimator = Pipeline(
+        steps=[
+            (
+                "regress",
+                ElasticNet(
+                    random_state=seed,
+                )
+            )
+        ]
+    )
+
+    cv_pipeline = GridSearchCV(
+        estimator=estimator,
+        param_grid=reg_parameters,
+        n_jobs=-1,
+        cv=n_folds,
+        scoring="neg_mean_squared_error",
+        return_train_score=True,
+    )
+
+    # Fit the model
+    cv_pipeline.fit(X=X_train, y=y_train.status)
+
+    # Obtain cross validation results
+    y_cv = cross_val_predict(
+        cv_pipeline.best_estimator_,
+        X=X_train,
+        y=y_train.status,
+        cv=n_folds,
+        method="predict",
+    )
+
+    # Get all performance results
+    y_predict_train = cv_pipeline.predict(X_train)
+    y_predict_test = cv_pipeline.predict(X_test)
+
+    return cv_pipeline, y_predict_train, y_predict_test, y_cv
+
+
+def get_preds(X_test_df, y_test_df, cv_pipeline, fold_no):
```

Comment:
This function seems very simple, but maybe a short description of what it does would be good.